### PR TITLE
fix(Checkbox): update stories and tests

### PR DESCRIPTION
**Pull request by laragoogle**
Tests are updated:
Screen Shot 2021-09-01 at 1 57 18 PM

### CODE REVIEW

File: packages/components/src/Form/Inputs/Checkbox/Checkbox.story.tsx [hunk 1]
```diff
@@ -57,44 +57,28 @@ export const Basic = Template.bind({
 
 export const Checked = Template.bind({})
 Checked.args = {
+  ...Basic,
```
Comment (laragoogle) [inline packages/components/src/Form/Inputs/Checkbox/Checkbox.story.tsx:60]: Some of the stories using Template were not displaying correctly. I couldn't find a solution for it that still use Template so I rewrote the stories.

File: packages/components/src/Form/Inputs/Checkbox/Checkbox.story.tsx [hunk 2]
```diff
@@ -57,44 +57,28 @@ export const Basic = Template.bind({
...
-  ...Checked.args,
-}
+export const ReadOnlyChecked = () => <Checkbox readOnly checked />
 
-export const ReadOnlyMixed = Template.bind({})
-ReadOnly.args = {
-  readOnly: true,
-  ...Mixed.args,
-}
+export const ReadOnlyCheckedMixed = () => <Checkbox readOnly checked="mixed" />
```
Comment (lukelooker) [inline packages/components/src/Form/Inputs/Checkbox/Checkbox.story.tsx:84]: Looking at the examples you replaced it looks like a lot of the `args` assignments we're mixed up:

image

`DisabledMixed` also has this scenario.

I checked out your branch and got it working with these CSF-style stories:

```

export const DisabledChecked = Template.bind({})
DisabledChecked.args = {
  ...Disabled.args,
}

export const DisabledCheckedMixed = Template.bind({})
DisabledCheckedMixed.args = {
  ...Disabled.args,
  ...MixedChecked.args,
}

export const ReadOnly = Template.bind({})
ReadOnly.args = {
  readOnly: true,
}

export const ReadOnlyChecked = Template.bind({})
ReadOnlyChecked.args = {
  ...Checked.args,
  readOnly: true,
}

export const ReadOnlyCheckedMixed = Template.bind({})
ReadOnlyCheckedMixed.args = {
  ...MixedChecked.args,
  readOnly: true,
}
```

File: packages/components/src/Form/Inputs/Checkbox/Checkbox.spec.tsx [hunk 3]
```diff
@@ -49,10 +50,45 @@ const runTimers = () =>
   })
 
 describe('Checkbox', () => {
+  test('renders', () => {
+    renderWithTheme(<Basic />)
+    expect(screen.getByRole('checkbox')).toBeInTheDocument()
+  })
+
+  test('renders checked', () => {
```
Comment (lukelooker) [inline packages/components/src/Form/Inputs/Checkbox/Checkbox.spec.tsx:58]:
```suggestion
  test('checked', () => {
```

File: packages/components/src/Form/Inputs/Checkbox/Checkbox.spec.tsx [hunk 4]
```diff
@@ -49,10 +50,45 @@ const runTimers = () =>
...
+    renderWithTheme(<Checked />)
+    const checkboxInput = screen.getByRole('checkbox')
+    expect(checkboxInput).toBeInTheDocument()
+    expect(checkboxInput as HTMLInputElement).toBeChecked()
+  })
+
+  test('renders checked mixed', () => {
```
Comment (lukelooker) [inline packages/components/src/Form/Inputs/Checkbox/Checkbox.spec.tsx:65]:
```suggestion
  test('mixed', () => {
```

File: packages/components/src/Form/Inputs/Checkbox/Checkbox.spec.tsx [hunk 5]
```diff
@@ -49,10 +50,45 @@ const runTimers = () =>
...
+    renderWithTheme(<MixedChecked />)
+    const checkboxInput = screen.getByRole('checkbox')
+
+    expect(checkboxInput).toBeInTheDocument()
+    expect(checkboxInput as HTMLInputElement).toBeChecked()
+    expect(screen.getByText('Check Mark Mixed')).toBeInTheDocument()
+  })
+
+  test('renders disabled', () => {
```
Comment (lukelooker) [inline packages/components/src/Form/Inputs/Checkbox/Checkbox.spec.tsx:74]:
```suggestion
  test('disabled', () => {
```

File: packages/components/src/Form/Inputs/Checkbox/Checkbox.spec.tsx [hunk 6]
```diff
@@ -49,10 +50,45 @@ const runTimers = () =>
...
+    renderWithTheme(<Disabled />)
+    const checkboxInput = screen.getByRole('checkbox')
+
+    expect(checkboxInput as HTMLInputElement).toBeDisabled()
+  })
+
+  test('renders disabled checked', () => {
```
Comment (lukelooker) [inline packages/components/src/Form/Inputs/Checkbox/Checkbox.spec.tsx:81]:
```suggestion
  test('disabled & checked', () => {
```

File: packages/components/src/Form/Inputs/Checkbox/Checkbox.spec.tsx [hunk 7]
```diff
@@ -49,10 +50,45 @@ const runTimers = () =>
...
+  test('renders', () => {
+    renderWithTheme(<Basic />)
+    expect(screen.getByRole('checkbox')).toBeInTheDocument()
+  })
+
+  test('renders checked', () => {
+    renderWithTheme(<Checked />)
+    const checkboxInput = screen.getByRole('checkbox')
+    expect(checkboxInput).toBeInTheDocument()
+    expect(checkboxInput as HTMLInputElement).toBeChecked()
```
Comment (lukelooker) [inline packages/components/src/Form/Inputs/Checkbox/Checkbox.spec.tsx:62]: `as HTMLInputElement` shouldn't be needed - poking at it locally tests appear to work without explicitly forcing the typing on these. Can you remove this explicit assignments?

```suggestion
    expect(checkboxInput).toBeChecked()
```

**Review by lukelooker — CHANGES_REQUESTED** [review]
The updated snapshots look good but I'd like to see the stories consistently use the CFS format - added a specific comment about what I suspect was going wrong there.

Thanks for adding additional unit test coverage along the way!

File: packages/components/src/Form/Inputs/Checkbox/Checkbox.story.tsx (hunk 2 above)
Comment (laragoogle) [reply packages/components/src/Form/Inputs/Checkbox/Checkbox.story.tsx:84]: oh no I can't believe I missed that 🤦🏽‍♀️

**Review by lukelooker — APPROVED** [review]
Looks good!

**Pull request merged**